### ISSUE TITLE
Remediate lists of links on Browse pages

### ISSUE THREAD

**Issue by erinchambers** [CDRH/orchid#329]
On `/browse` pages, the lists of links are tables with 3 columns: 1) # of items, 2) elipsis/separators, 3) filter/category. 

Here's an example from Whitman: https://whitmanarchive.org/browse/places?facet=places

Visually, the first 3 items in the list look like this:

[5](#)  [...........](#) [Boston (Mass.)](#)
[8](#)  [...........](#) [Brooklyn (New York, N.Y.)](#)
[16](#) [..........](#) [Camden, (N.J.)](#)

Screen reader output includes announces rows and columns, not just the links:

> Entering table.
>
> row 1 of 11 column 1 of 3
> link, 5
> column 2 of 3
> link, places
> column 3 of 3
> link, Boston, Mass
> 
> row 2 of 11 column 1 of 3
> link, 8
> column 2 of 3
> link, places
> column 3 of 3
> link, Brooklyn, New York, New York
>
> row 3 of 11 column 1 of 3
> link, 8
> column 2 of 3
> link, places
> column 3 of 3
> link, Camden, New Jersey

There are several problems with the current layout:
  1. The table is missing a caption and column headings, and we don't indicate anywhere that the #s in the first column represent the number of items matching that name/filter.. Either
    a. we should add a caption and table headings e.g. "Number of items" and the facet name (e.g. "Place"), or
    b. the table markup should be removed and the layout should be accomplished with an unordered list and then flexbox or grid
  2. The separator (`.......`) is read aloud to screen readers as a link. Furthermore, it is in its own column in the layout table, resulting in a lengthy and unnecessary announcement by screen readers ("column 2 of 3, link, {category name read from aria label, or URL if no aria label is present}")
  4. The number comes first and the name comes after, which might make the purpose of the table a bit confusing

# Suggested fixes

- [x] If the list really does function best as a table of data, add `<caption>` and appropriate table headings (`<th>`)
  - [x] If the separator stays, it should not be in its own column, and should be set up so screen readers don't read it aloud 
- [x] If the list should not be a table, discuss whether a single link per line is best, or whether # and name should remain separate links
  - [x] If the separator stays, create it with CSS or a pseudoelement and/or use `aria-hidden="true"` so screen readers don't read it aloud
- [x] Think about swapping the order of each row, with name first and # of items second

**Comment by erinchambers** [comment]
Discussed in Teams on 2025-09-24 - try a list of links, rather than a table, and do screen reader recordings of each (with NVDA and VoiceOver).

- [x] List of results as table
  - [ ] Recording of NVDA reading table
  - [ ] Recording of VoiceOver reading table
- [ ] List of results as unordered list
  - [ ] Recording of NVDA reading ul
  - [ ] Recording of VoiceOver reading ul
- [ ] Tabs as ARIA tabs
  - [ ] Recording of NVDA reading tabs
  - [ ] Recording of VoiceOver reading tabs
- [ ] Tabs as nav ul
  - [ ] Recording of NVDA reading nav ul
  - [ ] Recording of VoiceOver reading nav ul
